scitt/create_signed_statement.py

```diff
@@ -132,50 +123,88 @@ def main():
 
     parser = argparse.ArgumentParser(description="Create a signed statement.")
 
-    # signing key file
+    # content-type
     parser.add_argument(
-        "--signing-key-file",
+        "-t",
+        "--content-type",
         type=str,
-        help="filepath to the stored ecdsa P-256 signing key, in pem format.",
-        default="scitt-signing-key.pem",
+        help="The iana.org media type for the statement.",
+        default="application/json",
     )
 
-    # payload-file (a reference to the file that will become the payload of the SCITT Statement)
+    # detached-hash
     parser.add_argument(
-        "--payload-file",
+        "--detached-hash",
         type=str,
-        help="filepath to the content that will become the payload of the SCITT Statement "
-        "(currently limited to json format).",
-        default="scitt-payload.json",
+        help='The hash value to assist in payload verification when the payload-type="detached"'
     )
 
-    # content-type
+    # detached-hash-type
     parser.add_argument(
-        "--content-type",
+        "--detached-hash-type",
         type=str,
-        help="The iana.org media type for the payload",
-        default="application/json",
+        help='When the payload-type="detached", an optional detached-hash may be set to assist in payload verification. detached-hash-type identifies the hashing algorithm used'
     )
 
-    # feed
+    # issuer
     parser.add_argument(
-        "--feed",
+        "--issuer",
+        required=True,
         type=str,
-        help="feed to correlate statements made about an artifact.",
+        help="Owner of the signing key",
     )
 
-    # issuer
+    # location-hint
     parser.add_argument(
-        "--issuer",
+        "-l",
+        "--location-hint",
+        type=str,
+        help="An optional URI the statement is stored"
+    )
+
+    # statement-type
+    parser.add_argument(
+        "--payload-type",
+        type=str,
+        choices=['attached','detached','hash+sha256','hash+sha512'],
+        help='Signed Statements may attach the statement within the payload as attached, detached (nil), or sign a hash of the statement: (attached | detached | hash+[algo "hash+sha256" | "hash+sha512"])',
```

Comment:
I don't understand why this wouldn't just be [attached|detached]? 

If detached is specified, then `--detached-hash` and `--detached-hash-alg` are required argument?

Reply:
it may even be worth making these subcommands? or the non default a subcommand?

i.e.
`create_signed_statement detached --payload-hash=abcd --alg=sha256` Then we can remove the `detached` prefix from the options

Reply:
A subcommand is an interesting thought. 
However, all of these map to properties in the COSE headers, and those would be a flat structure.

Reply:
I fixed the parameter mixing. `payload-type` is now just `attached, detached`